src/DynamicRupture/FrictionLaws/GpuImpl/SlowVelocityWeakeningLaw.h

```diff
@@ -32,33 +32,38 @@ class SlowVelocityWeakeningLaw
   }
 
   struct MuDetails {
-    double a{};
-    double c{};
-    double ac{};
+    real a{};
+    real cLin{};
+    real cExpLog{};
+    real cExp{};
+    real acLin{};
   };
 
   SEISSOL_DEVICE static MuDetails getMuDetails(FrictionLawContext& ctx, double localStateVariable) {
-    const double localA = ctx.data->a[ctx.ltsFace][ctx.pointIndex];
+    const real localA = ctx.data->a[ctx.ltsFace][ctx.pointIndex];
     const double localSl0 = ctx.data->sl0[ctx.ltsFace][ctx.pointIndex];
     const double log1 = std::log(ctx.data->drParameters.rsSr0 * localStateVariable / localSl0);
-    const auto localF0 = ctx.data->f0[ctx.ltsFace][ctx.pointIndex];
-    const auto localB = ctx.data->b[ctx.ltsFace][ctx.pointIndex];
-    const double c =
-        0.5 / ctx.data->drParameters.rsSr0 * std::exp((localF0 + localB * log1) / localA);
-    return MuDetails{localA, c, localA * c};
+    const double localF0 = ctx.data->f0[ctx.ltsFace][ctx.pointIndex];
+    const double localB = ctx.data->b[ctx.ltsFace][ctx.pointIndex];
+
+    const real cLin = 0.5 / ctx.data->drParameters.rsSr0;
+    const real cExpLog = (localF0 + localB * log1) / localA;
```

Comment:
That's not the subject of the PR, but: wouldn't it make more sense to store one_over_localA and one_over_rsSr0? (or maybe getMuDetails is only called once?)

Reply:
Yes, that's only run once per Newton iteration. Since the state variable stays constant throughout a Newton iteration, we can pre-compute everything that only depends on the state variable and not on the slip rate to solve for.